apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/components/TreasuryStep/components/Parameter/Parameter.tsx

```diff
@@ -0,0 +1,245 @@
+import React, { FC, useCallback, useEffect, useMemo } from 'react';
+
+import { t } from 'i18next';
+
+import { SupportedTokenList, SupportedTokens } from '@sovryn/contracts';
+import {
+  FormGroup,
+  Input,
+  AmountInput,
+  Select,
+  Button,
+  ButtonStyle,
+  Paragraph,
+  ParagraphSize,
+  Icon,
+  IconNames,
+} from '@sovryn/ui';
+
+import { AssetRenderer } from '../../../../../../../../2_molecules/AssetRenderer/AssetRenderer';
+import { MaxButton } from '../../../../../../../../2_molecules/MaxButton/MaxButton';
+import { isAddress } from '../../../../../../../../3_organisms/StakeForm/components/AdjustStakeForm/AdjustStakeForm.utils';
+import { useAccount } from '../../../../../../../../../hooks/useAccount';
+import { useMaxAssetBalance } from '../../../../../../../../../hooks/useMaxAssetBalance';
+import { translations } from '../../../../../../../../../locales/i18n';
+import { useProposalContext } from '../../../../../../contexts/NewProposalContext';
+import { ProposalCreationParameter } from '../../../../../../contexts/ProposalContext.types';
+import { TREASURY_OPTIONS } from '../../TreasuryStep.constants';
+import { TreasuryParameterType } from '../../TreasuryStep.types';
+
+type ParameterProps = {
+  parameter: ProposalCreationParameter;
+  onError: (error: boolean) => void;
+};
+
+const pageTranslations = translations.bitocracyPage.proposalTreasuryForm;
+
+export const Parameter: FC<ParameterProps> = ({ parameter, onError }) => {
+  const { parameters, setParameters } = useProposalContext();
+  const { account } = useAccount();
+  const { balance: userBalance } = useMaxAssetBalance(
+    parameter.parametersStepExtraData?.token || SupportedTokens.rbtc,
+  );
```

Comment:
You need to show balance of the treasury, not user who submits proposal.

We should have a custom hook to receive that balance, because different contracts may have different approaches on how balance should be read.